[CDAP-18540] Allow levelDB compression to be enabled/disabled.

wyzhang opened the pull request:
Why:
When levelDB compression is enabled, LevelDB internal global serialization
could cause read performance to degrade substantially when there are high
concurrent reads/scans on the same levelDB. In such use cases, it would be
preferrable to disable compression given the suboptimal implementation
in java-version of levelDB

fernst reviewed (approved):
LGTM

chtyim commented on cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java:
```diff
@@ -1436,14 +1436,16 @@ public static String getZKQuorum(CConfiguration cConf) {
   public static final String CFG_DATA_INMEMORY_PERSISTENCE = "data.local.inmemory.persistence.type";
   public static final String CFG_DATA_LEVELDB_DIR = "data.local.storage";
-  public static final String CFG_DATA_LEVELDB_BLOCKSIZE = "data.local.storage.blocksize";
+  public static final String CFG_DATA_LEVELDB_BLOCKSIZE = "data.local.storage.compress";
```
call it `data.local.storage.compression.enabled`

wyzhang replied:
Fixed.

chtyim commented on cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java:
```diff
@@ -1436,14 +1436,16 @@ public static String getZKQuorum(CConfiguration cConf) {
    * Defaults for Data Fabric.
    */
   public static final String DEFAULT_DATA_LEVELDB_DIR = "data";
+  public static final boolean DEFAULT_DATA_LEVELDB_COMPRESS = true;
```
default shouldn't be declared in code. Move it into `cdap-default.xml`

chtyim commented on cdap-data-fabric/src/main/java/io/cdap/cdap/data2/dataset2/lib/table/leveldb/LevelDBTableService.java:
```diff
@@ -86,6 +88,7 @@ public void setConfiguration(CConfiguration config) {
     basePath = config.get(Constants.CFG_DATA_LEVELDB_DIR);
     Preconditions.checkNotNull(basePath, "No base directory configured for LevelDB.");
 
+    compress = config.getBoolean(Constants.CFG_DATA_LEVELDB_COMPRESS, Constants.DEFAULT_DATA_LEVELDB_COMPRESS);
```
Don't provide default in code. Use cdap-default.xml instead. The older code were not following the practice.

wyzhang replied:
Done.

wyzhang committed:
[CDAP-18540] Allow levelDB compression to be enabled/disabled.

Why:
When levelDB compression is enabled, LevelDB internal global serialization
could cause read performance to degrade substantially when there are high
concurrent reads/scans on the same levelDB. In such use cases, it would be
preferrable to disable compression given the suboptimal implementation
in java-version of levelDB
